### PR TITLE
Integrate Sigma Rules parser for importing rules based on Sigma schema-part2

### PR DESCRIPTION
**Pull request by sbcd90**
### Description
integrate sigma rules parser
 
### Issues Resolved
[Rule curation layer(P0) & Rule ingestion layer(P0)](https://github.com/opensearch-project/security-analytics/issues/5)
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/generated/org/opensearch/securityanalytics/rules/condition/Condition.g4 [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+start : expression;
+
+expression
+   : (SELECTOR|IDENTIFIER)                              # identOrSelectExpression
+   | LPAREN inner=expression RPAREN                     # parenExpression
+   | NOT expression                                     # notExpression
+   | left=expression operator=AND right=expression      # andExpression
+   | left=expression operator=OR right=expression       # orExpression
+   ;
```
[the remark this answers is not in the document]
Comment (sbcd90) [reply src/main/generated/org/opensearch/securityanalytics/rules/condition/Condition.g4]: fixed it

File: src/test/resources/fieldmappings.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+fieldmappings:
```
[the remark this answers is not in the document]
Comment (sbcd90) [reply src/test/resources/fieldmappings.yml]: fixed it. kept the path hardcoded in code. will generalize it as i add the logic for selecting mapping type.

**Review by getsaurabh02 — APPROVED** [review]

File: src/main/generated/org/opensearch/securityanalytics/rules/condition/ConditionLexer.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.TokenStream;
+import org.antlr.v4.runtime.*;
+import org.antlr.v4.runtime.atn.*;
+import org.antlr.v4.runtime.dfa.DFA;
+import org.antlr.v4.runtime.misc.*;
+
+@SuppressWarnings({"all", "warnings", "unchecked", "unused", "cast"})
+public class ConditionLexer extends Lexer {
+    static { RuntimeMetaData.checkVersion("4.10.1", RuntimeMetaData.VERSION); }
```
Comment (lezzago) [inline src/main/generated/org/opensearch/securityanalytics/rules/condition/ConditionLexer.java:18]: Should we have a constants file that stores this `4.10.1` version there so its easy to update?

File: src/main/java/org/opensearch/securityanalytics/rules/backend/OSQueryBackend.java [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+
+/*    @Override
+    public Object convertConditionFieldEqValNull(ConditionFieldEqualsValueExpression condition) {
+        return null;
+    }
+
+    @Override
+    public Object convertConditionFieldEqValQueryExpr(ConditionFieldEqualsValueExpression condition) {
+        return null;
+    }*/
```
Comment (lezzago) [inline src/main/java/org/opensearch/securityanalytics/rules/backend/OSQueryBackend.java:305]: Remove commented code. If we want to comment it temporarily, add a TODO here

File: src/main/java/org/opensearch/securityanalytics/rules/backend/OSQueryBackend.java [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+
+    @Override
+    public Object convertConditionValRe(ConditionValueExpression condition) {
+        return String.format(Locale.getDefault(), this.unboundReExpression, convertValueRe((SigmaRegularExpression) condition.getValue()));
+    }
+
+/*    @Override
+    public Object convertConditionValQueryExpr(ConditionValueExpression condition) {
+        return null;
+    }*/
```
Comment (lezzago) [inline src/main/java/org/opensearch/securityanalytics/rules/backend/OSQueryBackend.java:325]: Add TODO or remove

File: src/main/java/org/opensearch/securityanalytics/rules/backend/QueryBackend.java [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+            return this.convertConditionFieldEqValRe(condition);
+        } else if (condition.getValue() instanceof SigmaCIDRExpression) {
+            return this.convertConditionFieldEqValCidr(condition);
+        } else if (condition.getValue() instanceof SigmaCompareExpression) {
+            return this.convertConditionFieldEqValOpVal(condition);
+        } else if (condition.getValue() instanceof SigmaNull) {
+            return this.convertConditionFieldEqValNull(condition);
+        }/* else if (condition.getValue() instanceof SigmaQueryExpression) {
+            return this.convertConditionFieldEqValQueryExpr(condition);
+        }*/ else if (condition.getValue() instanceof SigmaExpansion) {
```
Comment (lezzago) [inline src/main/java/org/opensearch/securityanalytics/rules/backend/QueryBackend.java:186]: Add TODO or remove commented code

**Review by lezzago — COMMENTED** [review]
looks good, just have some minor comments

File: src/main/generated/org/opensearch/securityanalytics/rules/condition/ConditionLexer.java (hunk 3 above)
Comment (sbcd90) [reply src/main/generated/org/opensearch/securityanalytics/rules/condition/ConditionLexer.java:18]: this is in a generated class using antlr. hence, did not change it.

File: src/main/java/org/opensearch/securityanalytics/rules/backend/OSQueryBackend.java (hunk 4 above)
Comment (sbcd90) [reply src/main/java/org/opensearch/securityanalytics/rules/backend/OSQueryBackend.java:307]: fixed it.

File: src/main/java/org/opensearch/securityanalytics/rules/backend/OSQueryBackend.java (hunk 5 above)
Comment (sbcd90) [reply src/main/java/org/opensearch/securityanalytics/rules/backend/OSQueryBackend.java:329]: fixed it. added a TODO.

File: src/main/java/org/opensearch/securityanalytics/rules/backend/QueryBackend.java (hunk 6 above)
Comment (sbcd90) [reply src/main/java/org/opensearch/securityanalytics/rules/backend/QueryBackend.java]: fixed it. added a TODO.

**Review by lezzago — APPROVED** [review]

**Review by getsaurabh02 — APPROVED** [review]

**Pull request merged**